refract-spec.md

```diff
@@ -117,26 +295,334 @@ This example will show a JSON object represented as both a normal Refract Elemen
+    {
+      "element": "ref",
+      "content": "bar"
+    }
+  ]
+}
+```
+
+The resulting element would be the following. Note that the `id` was not included in the final element.
+
+```json
+{
+  "element": "foo",
+  "content": "second"
+}
+```
+
+### Select Element (Element)
+
+#### Properties
+
+- `element` select (string, fixed)
+- `content` (array[Option Element])
+
+#### Examples
+
+This example uses a Select Element to provide multiple options for the properites of an Object Element. One option is the value of the the key `firstName` is "John," and the other option is the value of the key `giveName` is "John."
+
+```json
+{
+  "element": "object",
+  "content": [
+    {
+      "element": "select",
+      "content": [
+        {
+          "element": "option",
+          "content": [
+            {
+              "element": "member",
+              "content": {
+                "key": {
+                  "element": "string",
+                  "content": "firstName"
+                },
+                "value": {
+                  "element": "string",
+                  "content": "John"
+                }
+              }
+            }
+          ]
+        },
+        {
+          "element": "option",
+          "content": [
+            {
+              "element": "member",
+              "content": {
+                "key": {
+                  "element": "string",
+                  "content": "givenName"
+                },
+                "value": {
+                  "element": "string",
+                  "content": "John"
+                }
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ]
 }
 ```
 
-##### Compact Element Representation
+For the above example, if the first option is chosen, the resulting object will look like this:
 
-This is how it would expressed in the compact version.
+```json
+{
+  "firstName": "John"
+}
+```
+
+If the second option is chosen, it will look like this:
 
 ```json
-["foo", {}, "bar"]
+{
+  "givenName": "John"
+}
 ```
 
+### Option Element (Element)
+
+The Option Element provides a way to give one or more elements that MAY be the value of the Select Element in which it is found. For examples, see the Select Element section.
+
+#### Properties
+
+- `element` option (string, fixed)
+- `content` (array[Element])
+
+## Defining New Elements
+
+When an element is given an ID, it creates a new type of element. When instances of the new element are used, they MAY conform to the definition of the referenced element.
+
+Elements do not inherit data from the referenced element, though they do inherit the definition. This means that a referenced element MAY define what an element MAY look like, but it does not define what it MUST look like, nor does it define data for each instance.
+
+When it is desired to inherit attributes, use the Extend Element and Ref Element together to accomplish this.
+
+### Example
+
+The element below is a string element with an ID of "foo."
```

Comment:
This is a big debate in Apiary right now :) The American way is to put periods inside the quotation marks.